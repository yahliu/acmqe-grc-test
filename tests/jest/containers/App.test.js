/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import App from '../../../src-web/containers/App'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
// import GrcApolloClient from '../../../lib/client/apollo-client'
// import { ApolloProvider } from 'react-apollo'
import { Provider } from 'react-redux'
import toJson from 'enzyme-to-json'

describe('App container test', () => {
  it('renders as expected', () => {
    const preloadedState = window.__PRELOADED_STATE__
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const match = {
      'path': '/multicloud',
      'url': '/multicloud',
      'isExact': false,
      'params': {}
    }
    const staticContext = {
      'title': 'IBM Multicloud Manager',
      'context': {
        'locale': 'en'
      },
      'xsrfToken': 'tXcZ8lkN-bDcRaa7cJmmIMX5EmsmJ_vI_YnU'
    }
    const component = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <App
            match={match}
            staticContext={staticContext}
          />
        </BrowserRouter>
      </Provider>
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
