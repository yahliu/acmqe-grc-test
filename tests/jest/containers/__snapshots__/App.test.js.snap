// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container test renders as expected 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <BrowserRouter>
    <_default
      match={
        Object {
          "isExact": false,
          "params": Object {},
          "path": "/multicloud/policies",
          "url": "/multicloud",
        }
      }
      url="/multicloud/policies"
    />
  </BrowserRouter>
</ContextProvider>
`;
